File: data/data.code.javascript.client/ReqTestXX.js
module.exports = ReqTestXX;

var Packet = require('../net/Packet');


function ReqTestXX() {
	this._id_u8 = 0;
	this._id_u16 = 0;
	this._id_u32 = 0;
	this._repeat_id_u8 = new Array();
	this._optional_id_u8_flag = 0;
	this._optional_id_u8 = undefined;


	this.Encode = function() {
		var packet = new Packet();
		packet.WriteByte(this._id_u8);
		packet.WriteUshort(this._id_u16);
		packet.WriteUint(this._id_u32);
		var repeat_id_u8_count = this._repeat_id_u8.length;
		packet.WriteUshort(repeat_id_u8_count);
		for (var i = 0; i < repeat_id_u8_count; i++)
		{
			var xxx = this._repeat_id_u8[i];
			packet.WriteByte(xxx);
		}
		packet.WriteByte(this._optional_id_u8_flag);
		if (this._optional_id_u8_flag == 1)
		{
			packet.WriteByte(this._optional_id_u8);
		}
		packet.Encode(40040);
		return packet;
	}
}

ReqTestXX.prototype = {
	set IdU8(val) {
		this._id_u8 = val;
	},
	get IdU8() {
		return this._id_u8;
	},

	set IdU16(val) {
		this._id_u16 = val;
	},
	get IdU16() {
		return this._id_u16;
	},

	set IdU32(val) {
		this._id_u32 = val;
	},
	get IdU32() {
		return this._id_u32;
	},

	set RepeatIdU8(val) {
		this._repeat_id_u8 = val;
	},
	get RepeatIdU8() {
		return this._repeat_id_u8;
	},

	set OptionalIdU8(val) {
		this._optional_id_u8_flag = 1;
		this._optional_id_u8 = val;
	},
	get OptionalIdU8() {
		return this._optional_id_u8;
	},
}

File: data/data.code.lua.client/ReqTestXX.lua
module("ReqTestXX", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqTestXX})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.id_u8 = 0
	self.id_u16 = 0
	self.id_u32 = 0
	self.repeat_id_u8 = {}
	self.optional_id_u8 = 0
end


function encode(self)
	local pack = packet.new()
	pack:writeU8(self.id_u8)
	pack:writeU16(self.id_u16)
	pack:writeU32(self.id_u32)
	local repeat_id_u8_count = table.getn(self.repeat_id_u8)
	pack:writeU16(repeat_id_u8_count)
	for i = 1, repeat_id_u8_count do
		pack:writeU8(table.remove(self.repeat_id_u8))
	end
	if self.optional_id_u8 ~= 0 then
		pack:writeU8(1)
		pack:writeU8(self.optional_id_u8)
	else
		pack:writeU8(0)
	end
	return pack:encode(Msg.P_REQ_TEST_X_X)
end

return ReqTestXX
